app/models/state_file_base_intake.rb

```diff
@@ -78,16 +78,21 @@ def self.selected_intakes_for_deadline_reminder_notifications
       .where(efile_submissions: { id: nil })
       .where.not(df_data_imported_at: nil)
       .has_verified_contact_info
-      .select(&:should_not_be_sent_reminder)
-  end
-
-  def should_not_be_sent_reminder
-    if message_tracker.present? && message_tracker["messages.state_file.finish_return"]
-      finish_return_msg_sent_time = Time.parse(message_tracker["messages.state_file.finish_return"])
-      finish_return_msg_sent_time < 24.hours.ago
-    else
-      !disqualifying_df_data_reason.present?
-    end
+      .select(&:should_be_sent_reminder)
+  end
+
+  def should_be_sent_reminder
+    received_reminder_recently = if message_tracker.present? && message_tracker["messages.state_file.finish_return"]
+                                   finish_return_msg_sent_time = Time.parse(message_tracker["messages.state_file.finish_return"])
+                                   finish_return_msg_sent_time > 24.hours.ago
+                                 else
+                                   false
+                                 end
+    matching_ssn_with_submission = self.class.where(hashed_ssn: hashed_ssn).excluding(self)
+    matching_email = matching_ssn_with_submission.where.not(email_address: nil).where(email_address: email_address)
+    matching_phone = matching_ssn_with_submission.where.not(phone_number: nil).where(phone_number: phone_number)
+    has_duplicate_with_submission = (matching_email + matching_phone).any? { |intake| intake.efile_submissions.any? }
```

Comment:
i kind of hate this please refactor if you have better ideas

Reply:
want to note that @mrotondo worked on a similar story to prevent matching hashed_ssn submissions & the difference between the two is that this PR cares whether the other intake has the same contact info while the other PR only cares if the intake has the same `hashed_ssn`.

Do we want to send duplicate messages for pre/post deadline reminders if a person has two intakes, both with same `hashed_ssn` but different contact info (email/phone_number), if one of the intakes have a submission? I think, probably not, but maybe we should clarify

Reply:
discussed in standup - will take over and add some refactoring commits to use the same logic from the above linked PR